var/spack/repos/builtin/packages/root/package.py

```diff
@@ -203,8 +205,10 @@ class Root(CMakePackage):
     # Qt4
     depends_on('qt@:4.999', when='+qt4')
 
-    # TMVA
+    # Python
+    depends_on('python@2.7:', when='+python', type=('build', 'run'))
     depends_on('py-numpy', when='+tmva')
+    depends_on('py-numpy', when='@6.20: +python')
```

Comment:
Is the `numpy` dependency now absolute? I'd thought there were options to build `tmva` in different ways.

Reply:
Fist of all: I am new to root, so I am happy about all sorts of help. So if you think, we need a new variant for customizing tmva, please elaborate.

That said:

My change is not related to tmva, but to root as a whole. And it seems to me, that `root+python` needs numpy. At least I wasn't able to build `root+python` without it.

I will retry soon.

Reply:
I've tried configuring 6.20.2 outside of spack, and the `tmva` option can live without `numpy`. There're a series of [CMake options to ROOT](https://github.com/root-project/root/blob/v6-20-02/cmake/modules/RootBuildOptions.cmake#L163) controlling TMVA options, so it looks like variants would be the best way to handle this, something like

```python
variant('tmva', default=True,
            description='Build TMVA multi variate analysis library')
variant('tmva-pymva', default=False,
            description='Enable python support in TMVA')

depends_on('py-numpy', when='+tmva-pymva')
conflicts('+tmva-pymva', '~tmva', 'tmva-pymva requires tmva')
```

That should work, but ROOT's CMake system can reactivate other options...

Reply:
Can we postpone this to a later PR?
To keep this PR focused on adding a new version?

Reply:
Could some of the confusion here perhaps be linked to https://sft.its.cern.ch/jira/browse/ROOT-10626 ?

Reply:
All of that looks very much like what happened to me, so YES.

What do you propose?

Reply:
Looks like the dependency on numpy was unintended, and will be fixed in 6.20.4 and 6.22. So you may want to add a FIXME comment clarifying that this dependency is accidental and eventually destined to go away, and therefore a suitable upper ROOT version bound should be added to this py-numpy dependency when the next ROOT update is added to this spackage.

Once that's done, I think this PR will be ready to go.

Reply:
Pushed new version with comment and `when='@6.20.00:6.20.03 …`.
Does that look good?

Reply:
Yup, I think that's good. Thanks for your patience!